### ISSUE TITLE
How to get the teacher model?

### ISSUE THREAD

**Issue by tonggege001** [bboylyg/NAD#2]
Hello, I have read your paper and learnt about NAD, but I have a question that how to get the teacher model? In your paper, it say "The teacher network can be obtained by an independent finetune process on the same clean data", but I have no idea how to "finetune". If you just finetune the last layer or all layers using clean data, I think it's hard to get a clean network, because the gradient of loss function in the clean data may be very low with convergency model

**Comment by tonggege001** [comment]
And I don't find the details in your code repo about teacher model finetune process. It would be grateful if you can share the  the finetune code of the teacher model.

**Comment by bboylyg** [comment]
Thanks for your interest in our work.  For your first question, we obtained the teacher model by finetuning all layers of the backdoored model using 5\% clean data with data augmentation techniques.  In our paper, we only finetune the backdoored model for 5~10 epochs. Please check more details of our experimental settings in section 4.1 and Appendix A; For the second question, the finetuning code is easy to get by just setting all the param Beta = 0, which means the distillation loss to be zero in the training process.

**Comment by tonggege001** [comment]
> Thanks for your interest in our work. For your first question, we obtained the teacher model by finetuning all layers of the backdoored model using 5% clean data with data augmentation techniques. In our paper, we only finetune the backdoored model for 5~10 epochs. Please check more details of our experimental settings in section 4.1 and Appendix A; For the second question, the finetuning code is easy to get by just setting all the param Beta = 0, which means the distillation loss to be zero in the training process.

Thank you very much!
